markdown-mode.el

```diff
@@ -8590,13 +8597,20 @@ position."
                (begin (and bounds (goto-char (nth 0 bounds)) (point-at-bol 2)))
                (end (and bounds (goto-char (nth 1 bounds)) (point-at-bol 1))))
           (if (and begin end)
-              (let* ((lang (markdown-code-block-lang))
+              (let* ((indentation (and (goto-char (nth 0 bounds)) (current-indentation)))
+                     (lang (markdown-code-block-lang))
                      (mode (or (and lang (markdown-get-lang-mode lang))
                                markdown-edit-code-block-default-mode))
                      (edit-indirect-guess-mode-function
                       (lambda (_parent-buffer _beg _end)
-                        (funcall mode))))
-                (edit-indirect-region begin end 'display-buffer))
+                        (funcall mode)))
+                     (indirect-buf (edit-indirect-region begin end 'display-buffer)))
+                (when (> indentation 0)
+                  (with-current-buffer indirect-buf
+                    (defvar markdown--code-block-indirect-indentation)
```

Comment:
Please declare global variable at top level and its default value `nil`

Reply:
Not sure if the variable should be declared toplevel since the scope itself is within the edit-indirect buffer, and not even really in markdown-mode itself. You sure this is best?